### PR TITLE
Update `flyteplugins`

### PR DESCRIPTION
**Pull request by bstadlbauer**
# TL;DR
Updates `flyteplugins` to the latest version. 

Removes all literal map hashing logic and replaces it with the one refactored into `flyteplugins` in https://github.com/flyteorg/flyteplugins/pull/363

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

### CODE REVIEW

**Comment by bstadlbauer** [comment]
@hamersaw Not quite sure what's failing the docker build step here 🤔 

I saw `k8s.io/kube-openapi` (indirect dependency) was updated, but not sure whether that's the issue

**Comment by hamersaw** [comment]
> @hamersaw Not quite sure what's failing the docker build step here thinking
> 
> I saw `k8s.io/kube-openapi` (indirect dependency) was updated, but not sure whether that's the issue

Seems to be an issue with [go version in the propeller Dockerfile](https://github.com/flyteorg/flytepropeller/blob/7b104d335d6fc41684217ce99781390cbd74cf84/Dockerfile#L6).

```
#11 79.12 /go/pkg/mod/k8s.io/kube-openapi@v0.0.0-20230501164219-8b0f38b5fd1f/pkg/cached/cache.go:242:16: undefined: atomic.Pointer
#11 79.12 note: module requires Go 1.19
```

I'll get this fixed.

File: pkg/controller/nodes/task/catalog/datacatalog/transformer.go [hunk 1]
```diff
@@ -116,72 +114,12 @@ func generateTaskSignatureHash(ctx context.Context, taskInterface core.TypedInte
...
-	}
-	hashifiedInputs := &core.LiteralMap{
-		Literals: hashifiedLiteralMap,
-	}
-
-	inputsHash, err := pbhash.ComputeHash(ctx, hashifiedInputs)
+	hashString, err := catalog.HashLiteralMap(ctx, inputs)
 	if err != nil {
-		return "", err
+		return "", nil
```
Comment (hamersaw) [inline pkg/controller/nodes/task/catalog/datacatalog/transformer.go:121]: Is returning `nil` here intended? Shouldn't this surface the error?

Comment (bstadlbauer) [reply pkg/controller/nodes/task/catalog/datacatalog/transformer.go]: Not intentional 🤦 Thanks for spotting, fixed in [4a19d66](https://github.com/flyteorg/flytepropeller/pull/578/commits/4a19d668fa21200112945ba32ff6b4a2bb65d4d6)

**Review by hamersaw — APPROVED** [review]

**Pull request merged**